libbeat/publisher/bc/publisher/pipeline.go

```diff
@@ -63,11 +54,27 @@ func createPipeline(
 		},
 	}
 
-	brokerFactory := func(e broker.Eventer) (broker.Broker, error) {
-		return membroker.NewBroker(e, queueSize, false), nil
+	brokerType := "mem"
+	if b := shipper.Queue.Name(); b != "" {
+		brokerType = b
 	}
 
-	p, err := pipeline.New(brokerFactory, out, settings)
+	brokerFactory := broker.FindFactory(brokerType)
```

Comment:
As quickly discussed, later we should probably rename `broker` package to `queue` to make the relation to the config more obvious.